.github/workflows/cd.yaml

```diff
@@ -30,4 +36,4 @@ jobs:
           git checkout main
           git pull
           git checkout $GITHUB_HEAD_REF
-          git pull
\ No newline at end of file
+          git pull
```

Comment:
So you just walked me through your code because I am not familiar with it. From what you explained, it looks like it should function according to what you said.